### PR TITLE
Adds throttling and corresponding tests

### LINKED ISSUE #2: Add throttling
**Is your feature request related to a problem? Please describe.**
We currently have basic rate-limiting provided by `express-limiter`. There seem to be other tools out in the open-source space, that have more advanced functionalities.

We do need throttling of requests, to balance out the rate, at which the back-end receives requests. That will also allow to mitigate pulse DDOS attacks, in which too many requests are sent, within a very short time window.

**Describe the solution you'd like**
- [ ] compare `express-limiter` to `node-rate-limiter` 
- [ ] add throttling
- [ ] add api tests showcasing throttling

**Describe alternatives you've considered**
- https://github.com/jhurliman/node-rate-limiter might be the better tool, as README provides explicit examples on throttling

### PR DESCRIPTION
**Pull request by agbilotia1998**
Resolves: #2 
- [x] Passed UAT

### CODE REVIEW

**Comment by agbilotia1998** [comment]
@michalmikolajczyk please review this PR, I've added the throttling retries using multiple modules, after testing of feature it seems that the required functionality is working. Please mention the changes if required. Thanks.

File: test/api/proxy.spec.js [hunk 1]
```diff
@@ -1,14 +1,40 @@
...
+process.env.BACK_END_URL = '';
+require('../../src/server');
 
 describe('reli-proxy', function module1() {
   this.timeout(10000);
+  /*
+  * Test the throttling
+  */
+  describe('/Testing the throttling of requests', () => {
+    it('it should rate limit second request', (done) => {
```
Comment (michalmikolajczyk) [inline test/api/proxy.spec.js]: nice test

**Review by michalmikolajczyk — CHANGES_REQUESTED** [review]
Looks like great work, but there are a few changes required.

- [ ] test showcasing throttling
- [ ] rate limiting with the option to limit by IP (that was present in the existing solution) OR all ips together

Also I wanted to ask, if the `RedisStore`, which is built using `rate-limit-redis`, should be used for both the `speedLimiter`, as the `rateLimiter`?

**Comment by michalmikolajczyk** [comment]
@agbilotia1998 I extend the duration of the Issue for 1 more week; it would be great, if we would be done earlier, obviously, but I did not want to imply expectation to work over the weekend. The work looks good so far, and you are welcome to take your time.

File: test/api/proxy.spec.js (hunk 1 above)
Comment (agbilotia1998) [reply test/api/proxy.spec.js]: @michalmikolajczyk yes, the Redis store is required for both speed limiting and rate-limiting as both the tasks require some kind of memory storage for keeping track of requests for IP, by default these module use MemoryStore.
1. Regarding the changes required, do you need a separate test just for rate limiting as the current test added shows both throttling with retry and rate limiting?
2. Currently, throttling is done by IP taken from `req.ip` and the same is stored as key in Redis store. Is some other key required here?
Thanks :)

**Comment by agbilotia1998** [comment]
Thanks, @michalmikolajczyk will complete it ASAP.

**Comment by agbilotia1998** [comment]
@michalmikolajczyk inferring from the requested changes, I've added the option to limit by IP or based on the number of requests from all IPs. Please review, thanks :)

**Review by michalmikolajczyk — CHANGES_REQUESTED** [review]
@agbilotia1998 this looks fine.

I believe we are missing tests showcasing throttling of requests – right now, we only have tests showing rate limiting with tokens (good test btw).

So in regard to throttling, please do either:
- add a test that displays throttling. That can be displayed with the use of a minimal custom back-end, to show the difference at request rate monitored at `reli-proxy`, and the back-end
- describe in README how to control rate limiting vs how to control throttling

**Comment by michalmikolajczyk** [comment]
@agbilotia1998 hey I reviewed the code again, it seems like your code does what's expected. Nice work!

**Review by michalmikolajczyk — APPROVED** [review]

**Pull request merged**